### PR TITLE
Cleaned up the span_where class method from Polycollections.

### PR DESCRIPTION
**Pull request by agautam478**
## PR summary

This PR removes the deprecated code during the version 3.7 from lib/matplotlib/collection.py which is a sub issue in the parent issues here: https://github.com/matplotlib/matplotlib/issues/26865

## PR checklist

- [ N/A] "closes #0000" is in the body of the PR description to [link the related issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)
- [ N/A] new and changed code is [tested](https://matplotlib.org/devdocs/devel/testing.html)
- [ N/A] *Plotting related* features are demonstrated in an [example](https://matplotlib.org/devdocs/devel/document.html#write-examples-and-tutorials)
- [ N/A] *New Features* and *API Changes* are noted with a [directive and release note](https://matplotlib.org/devdocs/devel/coding_guide.html#new-features-and-api-changes)
- [ N/A] Documentation complies with [general](https://matplotlib.org/devdocs/devel/document.html#write-rest-pages) and [docstring](https://matplotlib.org/devdocs/devel/document.html#write-docstrings) guidelines

### CODE REVIEW

File: doc/api/next_api_changes/removals/26874-AG.rst [hunk 1]
```diff
@@ -0,0 +1,4 @@
+``collections.PolyCollection.span_where``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
Comment (QuLogic) [inline doc/api/next_api_changes/removals/26874-AG.rst:2]: Also include `BrokenBarHCollection`
```suggestion
``collections.PolyCollection.span_where`` and ``collections.BrokenBarHCollection``
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

File: doc/api/next_api_changes/removals/26874-AG.rst [hunk 2]
```diff
@@ -0,0 +1,4 @@
...
+
+... removed as it was deprecated during 3.8. Use ``fill_between`` instead
```
Comment (QuLogic) [inline doc/api/next_api_changes/removals/26874-AG.rst:4]:
```suggestion
... removed as it was deprecated during 3.7. Use ``fill_between`` instead
```

File: lib/matplotlib/collections.py [hunk 3]
```diff
@@ -2344,9 +2313,9 @@ def set_array(self, A):
         # elements and thus compression occurring.
         if self._deprecated_compression and np.ndim(A) == 1:
             _api.warn_deprecated("3.8", message="Setting a PolyQuadMesh array using "
-                                                "the compressed values is deprecated. "
-                                                "Pass the full 2D shape of the original array "
-                                                f"{prev_unmask.shape} including the masked elements.")
+                                 "the compressed values is deprecated. "
```
Comment (tacaswell) [inline lib/matplotlib/collections.py:2316]: Were these changes intentional?

Comment (agautam478) [reply lib/matplotlib/collections.py]: The pep8 linter fixed the sapcings.

**Comment by QuLogic** [comment]
Linting shows that there is a little more fallout from this removal:
1. An import of `BrokenBarHCollection` in `lib/matplotlib/axes/_axes.pyi` and `lib/matplotlib/pyplot.py`
2. A doc reference in `doc/users/prev_whats_new/whats_new_1.4.rst`; this should be changed to code style (remove the `:class:` and change the single back ticks to double backticks).
3. In `lib/matplotlib/axes/_axes.pyi` and `lib/matplotlib/pyplot.py`, `broken_barh` is typed as returning `BrokenBarHCollection`; this is _wrong_ and should be changed to `PolyCollection`.

**Comment by QuLogic** [comment]
Sorry, there is one more remaining instance in the docs at `doc/api/artist_api.rst`.

**Comment by agautam478** [comment]
yayy! finally the build passed. 👯

File: doc/users/prev_whats_new/whats_new_1.4.rst [hunk 4]
```diff
@@ -221,7 +221,6 @@ Added size related functions to specialized `.Collection`\s
 Added the ``get_size`` and ``set_size`` functions to control the size of
 elements of specialized collections (
 :class:`~matplotlib.collections.AsteriskPolygonCollection`
-:class:`~matplotlib.collections.BrokenBarHCollection`
```
Comment (QuLogic) [inline doc/users/prev_whats_new/whats_new_1.4.rst:224]: This shouldn't be removed from old docs, even though it doesn't exist any more, because it's a historical reflection of "what's new in 1.4". Mark it as code style instead:
```suggestion
``matplotlib.collections.BrokenBarHCollection``
```

**Review by QuLogic — APPROVED** [review]
If you are comfortable rebasing and squashing this down to one commit, please do so. Otherwise, I suggest the person who merges do so.

**Review by rcomer — APPROVED** [review]
